calyx-py/calyx/builder.py

```diff
@@ -268,53 +277,64 @@ def seq_mem_d1(
             name, ast.Stdlib.seq_mem_d1(bitwidth, len, idx_size), is_external, is_ref
         )
 
-    def add(self, name: str, size: int, signed=False) -> CellBuilder:
-        """Generate a StdAdd cell."""
+    def binary_op_helper(
+        self,
+        operation: str,
+        size: int,
+        name=None,
+        signed=False,
+    ) -> CellBuilder:
+        """Generate a binary cell of the kind specified in {operation}."""
         self.prog.import_("primitives/binary_operators.futil")
-        return self.cell(name, ast.Stdlib.op("add", size, signed))
+        name = name or f"{operation}_{self.generate_name(operation)}"
```

Comment:
Should this perhaps just be `name = name or self.generate_name(operation)}`? As it stands, it seems like it would generate names like `eq_eq_0`.

Reply:
Thanks for the catch!